cluster-autoscaler/core/scaleup/orchestrator/executor_test.go

```diff
@@ -0,0 +1,127 @@
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package orchestrator
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	"k8s.io/autoscaler/cluster-autoscaler/utils/errors"
+)
+
+func TestCombinedConcurrentScaleUpErrors(t *testing.T) {
+	cloudProviderErr := errors.NewAutoscalerError(errors.CloudProviderError, "provider error")
+	internalErr := errors.NewAutoscalerError(errors.InternalError, "internal error")
+	testCases := []struct {
+		desc        string
+		errors      []errors.AutoscalerError
+		expectedErr errors.AutoscalerError
+	}{
+		{
+			desc:        "no errors",
+			errors:      []errors.AutoscalerError{},
+			expectedErr: nil,
+		},
+		{
+			desc:        "single error",
+			errors:      []errors.AutoscalerError{internalErr},
+			expectedErr: internalErr,
+		},
+		{
+			desc: "two duplicated errors",
+			errors: []errors.AutoscalerError{
+				internalErr,
+				internalErr,
+			},
+			expectedErr: internalErr,
+		},
+		{
+			desc: "two different errors",
+			errors: []errors.AutoscalerError{
+				cloudProviderErr,
+				internalErr,
+			},
+			expectedErr: errors.NewAutoscalerError(
+				errors.CloudProviderError,
+				"provider error ...other concurrent errors: [\"[internalError] internal error\"]",
+			),
+		},
+		{
+			desc: "two different errors - reverse alphabetical order",
+			errors: []errors.AutoscalerError{
+				internalErr,
+				cloudProviderErr,
+			},
+			expectedErr: errors.NewAutoscalerError(
+				errors.CloudProviderError,
+				"provider error ...other concurrent errors: [\"[internalError] internal error\"]",
+			),
+		},
+		{
+			desc: "errors with the same type and different messages",
+			errors: []errors.AutoscalerError{
+				errors.NewAutoscalerError(errors.InternalError, "A"),
+				errors.NewAutoscalerError(errors.InternalError, "B"),
+				errors.NewAutoscalerError(errors.InternalError, "C"),
+			},
+			expectedErr: errors.NewAutoscalerError(
+				errors.InternalError,
+				"A ...other concurrent errors: [\"B\", \"C\"]"),
+		},
+		{
+			desc: "errors with the same type and some duplicated messages",
+			errors: []errors.AutoscalerError{
+				errors.NewAutoscalerError(errors.InternalError, "A"),
+				errors.NewAutoscalerError(errors.InternalError, "B"),
+				errors.NewAutoscalerError(errors.InternalError, "A"),
+			},
+			expectedErr: errors.NewAutoscalerError(
+				errors.InternalError,
+				"A ...other concurrent errors: [\"B\"]"),
+		},
+		{
+			desc: "some duplicated errors",
+			errors: []errors.AutoscalerError{
+				errors.NewAutoscalerError(errors.CloudProviderError, "A"),
+				errors.NewAutoscalerError(errors.CloudProviderError, "A"),
+				errors.NewAutoscalerError(errors.CloudProviderError, "B"),
+				errors.NewAutoscalerError(errors.InternalError, "A"),
+			},
+			expectedErr: errors.NewAutoscalerError(
+				errors.CloudProviderError,
+				"A ...other concurrent errors: [\"[cloudProviderError] B\", \"[internalError] A\"]"),
```

Comment:
This seems non-intuitive that we skip the type (`CloudProviderError`) for the first error `A`, but print it for other errors.
The `...` looks rather artificial. 
Also this message is losing the information on which node group we encountered the given error, which I think is quite valuable, especially when debugging.